In [65]:
import pandas as pd

In [67]:
file=pd.read_csv("C:/Users/Asus/Documents/data analyst/Excel files/Outlet.csv")

In [69]:
file.head() 

Item_Identifier  Item_Weight Item_Fat_Content  Item_Visibility  \
0           FDA15         9.30          Low Fat         0.016047   
1           DRC01         5.92          Regular         0.019278   
2           FDN15        17.50          Low Fat         0.016760   
3           FDX07        19.20          Regular         0.000000   
4           NCD19         8.93          Low Fat         0.000000   

               Item_Type  Item_MRP Outlet_Identifier  \
0                  Dairy  249.8092            OUT049   
1            Soft Drinks   48.2692            OUT018   
2                   Meat  141.6180            OUT049   
3  Fruits and Vegetables  182.0950            OUT010   
4              Household   53.8614            OUT013   

   Outlet_Establishment_Year Outlet_Size Outlet_Location_Type  \
0                       1999      Medium               Tier 1   
1                       2009      Medium               Tier 3   
2                       1999      Medium               Tier 1   
3                       1998         NaN               Tier 3   
4                       1987        High               Tier 3   

         Outlet_Type  Item_Outlet_Sales  
0  Supermarket Type1          3735.1380  
1  Supermarket Type2           443.4228  
2  Supermarket Type1          2097.2700  
3      Grocery Store           732.3800  
4  Supermarket Type1           994.7052

***head function helps to print the first five rows of the file you inserted in jupiter.***

In [72]:
file.tail() 

Item_Identifier  Item_Weight Item_Fat_Content  Item_Visibility  \
8518           FDF22        6.865          Low Fat         0.056783   
8519           FDS36        8.380          Regular         0.046982   
8520           NCJ29       10.600          Low Fat         0.035186   
8521           FDN46        7.210          Regular         0.145221   
8522           DRG01       14.800          Low Fat         0.044878   

               Item_Type  Item_MRP Outlet_Identifier  \
8518         Snack Foods  214.5218            OUT013   
8519        Baking Goods  108.1570            OUT045   
8520  Health and Hygiene   85.1224            OUT035   
8521         Snack Foods  103.1332            OUT018   
8522         Soft Drinks   75.4670            OUT046   

      Outlet_Establishment_Year Outlet_Size Outlet_Location_Type  \
8518                       1987        High               Tier 3   
8519                       2002         NaN               Tier 2   
8520                       2004       Small               Tier 2   
8521                       2009      Medium               Tier 3   
8522                       1997       Small               Tier 1   

            Outlet_Type  Item_Outlet_Sales  
8518  Supermarket Type1          2778.3834  
8519  Supermarket Type1           549.2850  
8520  Supermarket Type1          1193.1136  
8521  Supermarket Type2          1845.5976  
8522  Supermarket Type1           765.6700

 ***the tail function helps to print the last five rows of the excel file you inserted in jupiter.***

In [75]:
file.shape # 

(8523, 12)

***the shape function helps to find out the total number of rows and columns in your excel sheet.***

 # the output shown above refers that there are 8523 number of rows and 12 columns.
           

In [79]:
file.info() 

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8523 entries, 0 to 8522
Data columns (total 12 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   Item_Identifier            8523 non-null   object 
 1   Item_Weight                7060 non-null   float64
 2   Item_Fat_Content           8523 non-null   object 
 3   Item_Visibility            8523 non-null   float64
 4   Item_Type                  8523 non-null   object 
 5   Item_MRP                   8523 non-null   float64
 6   Outlet_Identifier          8523 non-null   object 
 7   Outlet_Establishment_Year  8523 non-null   int64  
 8   Outlet_Size                6113 non-null   object 
 9   Outlet_Location_Type       8523 non-null   object 
 10  Outlet_Type                8523 non-null   object 
 11  Item_Outlet_Sales          8523 non-null   float64
dtypes: float64(4), int64(1), object(7)
memory usage: 799.2+ KB


## info method prints information about the data frame like numbers of column,data type ,memory usage etc.

In [82]:
file.describe() 

Item_Weight  Item_Visibility     Item_MRP  Outlet_Establishment_Year  \
count  7060.000000      8523.000000  8523.000000                8523.000000   
mean     12.857645         0.066132   140.992782                1997.831867   
std       4.643456         0.051598    62.275067                   8.371760   
min       4.555000         0.000000    31.290000                1985.000000   
25%       8.773750         0.026989    93.826500                1987.000000   
50%      12.600000         0.053931   143.012800                1999.000000   
75%      16.850000         0.094585   185.643700                2004.000000   
max      21.350000         0.328391   266.888400                2009.000000   

       Item_Outlet_Sales  
count        8523.000000  
mean         2181.288914  
std          1706.499616  
min            33.290000  
25%           834.247400  
50%          1794.331000  
75%          3101.296400  
max         13086.964800

# describe method helps to  get a overview of data.

In [85]:
file.columns 

Index(['Item_Identifier', 'Item_Weight', 'Item_Fat_Content', 'Item_Visibility',
       'Item_Type', 'Item_MRP', 'Outlet_Identifier',
       'Outlet_Establishment_Year', 'Outlet_Size', 'Outlet_Location_Type',
       'Outlet_Type', 'Item_Outlet_Sales'],
      dtype='object')

## It helps to know all the names of the column,and also you can copy the column further for any other use so that you won't find any error regarding uppercase or lowercase as we know that python is a casesensitive language.

In [88]:
file.set_index(['Item_Identifier','Outlet_Identifier']) 

Item_Weight Item_Fat_Content  \
Item_Identifier Outlet_Identifier                                 
FDA15           OUT049                   9.300          Low Fat   
DRC01           OUT018                   5.920          Regular   
FDN15           OUT049                  17.500          Low Fat   
FDX07           OUT010                  19.200          Regular   
NCD19           OUT013                   8.930          Low Fat   
...                                        ...              ...   
FDF22           OUT013                   6.865          Low Fat   
FDS36           OUT045                   8.380          Regular   
NCJ29           OUT035                  10.600          Low Fat   
FDN46           OUT018                   7.210          Regular   
DRG01           OUT046                  14.800          Low Fat   

                                   Item_Visibility              Item_Type  \
Item_Identifier Outlet_Identifier                                           
FDA15           OUT049                    0.016047                  Dairy   
DRC01           OUT018                    0.019278            Soft Drinks   
FDN15           OUT049                    0.016760                   Meat   
FDX07           OUT010                    0.000000  Fruits and Vegetables   
NCD19           OUT013                    0.000000              Household   
...                                            ...                    ...   
FDF22           OUT013                    0.056783            Snack Foods   
FDS36           OUT045                    0.046982           Baking Goods   
NCJ29           OUT035                    0.035186     Health and Hygiene   
FDN46           OUT018                    0.145221            Snack Foods   
DRG01           OUT046                    0.044878            Soft Drinks   

                                   Item_MRP  Outlet_Establishment_Year  \
Item_Identifier Outlet_Identifier                                        
FDA15           OUT049             249.8092                       1999   
DRC01           OUT018              48.2692                       2009   
FDN15           OUT049             141.6180                       1999   
FDX07           OUT010             182.0950                       1998   
NCD19           OUT013              53.8614                       1987   
...                                     ...                        ...   
FDF22           OUT013             214.5218                       1987   
FDS36           OUT045             108.1570                       2002   
NCJ29           OUT035              85.1224                       2004   
FDN46           OUT018             103.1332                       2009   
DRG01           OUT046              75.4670                       1997   

                                  Outlet_Size Outlet_Location_Type  \
Item_Identifier Outlet_Identifier                                    
FDA15           OUT049                 Medium               Tier 1   
DRC01           OUT018                 Medium               Tier 3   
FDN15           OUT049                 Medium               Tier 1   
FDX07           OUT010                    NaN               Tier 3   
NCD19           OUT013                   High               Tier 3   
...                                       ...                  ...   
FDF22           OUT013                   High               Tier 3   
FDS36           OUT045                    NaN               Tier 2   
NCJ29           OUT035                  Small               Tier 2   
FDN46           OUT018                 Medium               Tier 3   
DRG01           OUT046                  Small               Tier 1   

                                         Outlet_Type  Item_Outlet_Sales  
Item_Identifier Outlet_Identifier                                        
FDA15           OUT049             Supermarket Type1          3735.1380  
DRC01           OUT018             Supermarket Type2           44

# the set_index function helps to set one or more existing columns as its index.#

In [91]:
file.loc[0:100:1,['Item_Identifier','Item_Fat_Content','Item_Type','Item_MRP']]

Item_Identifier Item_Fat_Content              Item_Type  Item_MRP
0             FDA15          Low Fat                  Dairy  249.8092
1             DRC01          Regular            Soft Drinks   48.2692
2             FDN15          Low Fat                   Meat  141.6180
3             FDX07          Regular  Fruits and Vegetables  182.0950
4             NCD19          Low Fat              Household   53.8614
..              ...              ...                    ...       ...
96            FDF41          Low Fat           Frozen Foods  246.0460
97            FDB56          Regular  Fruits and Vegetables  187.4556
98            NCP18          Low Fat              Household  149.9708
99            FDB56          Regular  Fruits and Vegetables  187.8556
100           FDT28          Low Fat           Frozen Foods  151.0708

[101 rows x 4 columns]

## the above loc function is used to select specify rows and columns from dataFrame.

In [94]:
file.iloc[1:101]

Item_Identifier  Item_Weight Item_Fat_Content  Item_Visibility  \
1             DRC01        5.920          Regular         0.019278   
2             FDN15       17.500          Low Fat         0.016760   
3             FDX07       19.200          Regular         0.000000   
4             NCD19        8.930          Low Fat         0.000000   
5             FDP36       10.395          Regular         0.000000   
..              ...          ...              ...              ...   
96            FDF41       12.150          Low Fat         0.131384   
97            FDB56        8.750          Regular         0.074613   
98            NCP18          NaN          Low Fat         0.028460   
99            FDB56        8.750          Regular         0.074627   
100           FDT28       13.300          Low Fat         0.063695   

                 Item_Type  Item_MRP Outlet_Identifier  \
1              Soft Drinks   48.2692            OUT018   
2                     Meat  141.6180            OUT049   
3    Fruits and Vegetables  182.0950            OUT010   
4                Household   53.8614            OUT013   
5             Baking Goods   51.4008            OUT018   
..                     ...       ...               ...   
96            Frozen Foods  246.0460            OUT049   
97   Fruits and Vegetables  187.4556            OUT035   
98               Household  149.9708            OUT027   
99   Fruits and Vegetables  187.8556            OUT046   
100           Frozen Foods  151.0708            OUT045   

     Outlet_Establishment_Year Outlet_Size Outlet_Location_Type  \
1                         2009      Medium               Tier 3   
2                         1999      Medium               Tier 1   
3                         1998         NaN               Tier 3   
4                         1987        High               Tier 3   
5                         2009      Medium               Tier 3   
..                         ...         ...                  ...   
96                        1999      Medium               Tier 1   
97                        2004       Small               Tier 2   
98                        1985      Medium               Tier 3   
99                        1997       Small               Tier 1   
100                       2002         NaN               Tier 2   

           Outlet_Type  Item_Outlet_Sales  
1    Supermarket Type2           443.4228  
2    Supermarket Type1          2097.2700  
3        Grocery Store           732.3800  
4    Supermarket Type1           994.7052  
5    Supermarket Type2           556.6088  
..                 ...                ...  
96   Supermarket Type1          1231.7300  
97   Supermarket Type1          3755.1120  
98   Supermarket Type3          4363.6532  
99   Supermarket Type1          1314.2892  
100  Supermarket Type1          1805.6496  

[100 rows x 12 columns]

## iloc function helps user to select particular rows and columns of dataframe by their integer position

In [97]:
file.drop('Item_Outlet_Sales',axis=1)

Item_Identifier  Item_Weight Item_Fat_Content  Item_Visibility  \
0              FDA15        9.300          Low Fat         0.016047   
1              DRC01        5.920          Regular         0.019278   
2              FDN15       17.500          Low Fat         0.016760   
3              FDX07       19.200          Regular         0.000000   
4              NCD19        8.930          Low Fat         0.000000   
...              ...          ...              ...              ...   
8518           FDF22        6.865          Low Fat         0.056783   
8519           FDS36        8.380          Regular         0.046982   
8520           NCJ29       10.600          Low Fat         0.035186   
8521           FDN46        7.210          Regular         0.145221   
8522           DRG01       14.800          Low Fat         0.044878   

                  Item_Type  Item_MRP Outlet_Identifier  \
0                     Dairy  249.8092            OUT049   
1               Soft Drinks   48.2692            OUT018   
2                      Meat  141.6180            OUT049   
3     Fruits and Vegetables  182.0950            OUT010   
4                 Household   53.8614            OUT013   
...                     ...       ...               ...   
8518            Snack Foods  214.5218            OUT013   
8519           Baking Goods  108.1570            OUT045   
8520     Health and Hygiene   85.1224            OUT035   
8521            Snack Foods  103.1332            OUT018   
8522            Soft Drinks   75.4670            OUT046   

      Outlet_Establishment_Year Outlet_Size Outlet_Location_Type  \
0                          1999      Medium               Tier 1   
1                          2009      Medium               Tier 3   
2                          1999      Medium               Tier 1   
3                          1998         NaN               Tier 3   
4                          1987        High               Tier 3   
...                         ...         ...                  ...   
8518                       1987        High               Tier 3   
8519                       2002         NaN               Tier 2   
8520                       2004       Small               Tier 2   
8521                       2009      Medium               Tier 3   
8522                       1997       Small               Tier 1   

            Outlet_Type  
0     Supermarket Type1  
1     Supermarket Type2  
2     Supermarket Type1  
3         Grocery Store  
4     Supermarket Type1  
...                 ...  
8518  Supermarket Type1  
8519  Supermarket Type1  
8520  Supermarket Type1  
8521  Supermarket Type2  
8522  Supermarket Type1  

[8523 rows x 11 columns]

## Drop() method helps to remove particular row or column

In [100]:
file[((file['Item_MRP']>200)& (file['Item_MRP']<=250)& (file['Outlet_Size']=='Medium'))]

Item_Identifier  Item_Weight Item_Fat_Content  Item_Visibility  \
0              FDA15        9.300          Low Fat         0.016047   
55             FDK21        7.905          Low Fat         0.010053   
60             FDM20       10.000          Low Fat         0.000000   
96             FDF41       12.150          Low Fat         0.131384   
104            FDP09       19.750          Low Fat         0.034028   
...              ...          ...              ...              ...   
8347           FDB10       10.000          Low Fat         0.067313   
8356           FDN45          NaN          Low Fat         0.117531   
8451           FDK21        7.905          Low Fat         0.010028   
8491           FDO03       10.395          Regular         0.037033   
8505           FDB46       10.500          Regular         0.094146   

                  Item_Type  Item_MRP Outlet_Identifier  \
0                     Dairy  249.8092            OUT049   
55              Snack Foods  249.0408            OUT018   
60    Fruits and Vegetables  246.9144            OUT018   
96             Frozen Foods  246.0460            OUT049   
104             Snack Foods  212.0902            OUT018   
...                     ...       ...               ...   
8347            Snack Foods  234.3590            OUT049   
8356            Snack Foods  222.7088            OUT027   
8451            Snack Foods  249.6408            OUT049   
8491                   Meat  227.9352            OUT018   
8505            Snack Foods  210.8244            OUT018   

      Outlet_Establishment_Year Outlet_Size Outlet_Location_Type  \
0                          1999      Medium               Tier 1   
55                         2009      Medium               Tier 3   
60                         2009      Medium               Tier 3   
96                         1999      Medium               Tier 1   
104                        2009      Medium               Tier 3   
...                         ...         ...                  ...   
8347                       1999      Medium               Tier 1   
8356                       1985      Medium               Tier 3   
8451                       1999      Medium               Tier 1   
8491                       2009      Medium               Tier 3   
8505                       2009      Medium               Tier 3   

            Outlet_Type  Item_Outlet_Sales  
0     Supermarket Type1          3735.1380  
55    Supermarket Type2          6258.5200  
60    Supermarket Type2          3185.1872  
96    Supermarket Type1          1231.7300  
104   Supermarket Type2          3185.8530  
...                 ...                ...  
8347  Supermarket Type1          3072.6670  
8356  Supermarket Type3          4026.7584  
8451  Supermarket Type1          3254.4304  
8491  Supermarket Type2          4809.7392  
8505  Supermarket Type2          2117.2440  

[327 rows x 12 columns]

In [102]:
file[file.isin({'Outlet_Size':['High','Medium']})]

Item_Identifier  Item_Weight Item_Fat_Content  Item_Visibility Item_Type  \
0                NaN          NaN              NaN              NaN       NaN   
1                NaN          NaN              NaN              NaN       NaN   
2                NaN          NaN              NaN              NaN       NaN   
3                NaN          NaN              NaN              NaN       NaN   
4                NaN          NaN              NaN              NaN       NaN   
...              ...          ...              ...              ...       ...   
8518             NaN          NaN              NaN              NaN       NaN   
8519             NaN          NaN              NaN              NaN       NaN   
8520             NaN          NaN              NaN              NaN       NaN   
8521             NaN          NaN              NaN              NaN       NaN   
8522             NaN          NaN              NaN              NaN       NaN   

      Item_MRP Outlet_Identifier  Outlet_Establishment_Year Outlet_Size  \
0          NaN               NaN                        NaN      Medium   
1          NaN               NaN                        NaN      Medium   
2          NaN               NaN                        NaN      Medium   
3          NaN               NaN                        NaN         NaN   
4          NaN               NaN                        NaN        High   
...        ...               ...                        ...         ...   
8518       NaN               NaN                        NaN        High   
8519       NaN               NaN                        NaN         NaN   
8520       NaN               NaN                        NaN         NaN   
8521       NaN               NaN                        NaN      Medium   
8522       NaN               NaN                        NaN         NaN   

     Outlet_Location_Type Outlet_Type  Item_Outlet_Sales  
0                     NaN         NaN                NaN  
1                     NaN         NaN                NaN  
2                     NaN         NaN                NaN  
3                     NaN         NaN                NaN  
4                     NaN         NaN                NaN  
...                   ...         ...                ...  
8518                  NaN         NaN                NaN  
8519                  NaN         NaN                NaN  
8520                  NaN         NaN                NaN  
8521                  NaN         NaN                NaN  
8522                  NaN         NaN                NaN  

[8523 rows x 12 columns]

## The isin() method checks if the Dataframe contains the specified values.

In [105]:
def Item_Outlet_sales(x):
    if x<=1500:
        return 'low'
    if x>1500 and x<6000:
        return 'medium'
    else:
        return 'high'

In [107]:
file['sales_category']=file['Item_Outlet_Sales'].apply(Item_Outlet_sales)
file.head()

Item_Identifier  Item_Weight Item_Fat_Content  Item_Visibility  \
0           FDA15         9.30          Low Fat         0.016047   
1           DRC01         5.92          Regular         0.019278   
2           FDN15        17.50          Low Fat         0.016760   
3           FDX07        19.20          Regular         0.000000   
4           NCD19         8.93          Low Fat         0.000000   

               Item_Type  Item_MRP Outlet_Identifier  \
0                  Dairy  249.8092            OUT049   
1            Soft Drinks   48.2692            OUT018   
2                   Meat  141.6180            OUT049   
3  Fruits and Vegetables  182.0950            OUT010   
4              Household   53.8614            OUT013   

   Outlet_Establishment_Year Outlet_Size Outlet_Location_Type  \
0                       1999      Medium               Tier 1   
1                       2009      Medium               Tier 3   
2                       1999      Medium               Tier 1   
3                       1998         NaN               Tier 3   
4                       1987        High               Tier 3   

         Outlet_Type  Item_Outlet_Sales sales_category  
0  Supermarket Type1          3735.1380         medium  
1  Supermarket Type2           443.4228            low  
2  Supermarket Type1          2097.2700         medium  
3      Grocery Store           732.3800            low  
4  Supermarket Type1           994.7052            low

## The apply() method allows user to apply a function along one of the axis of the Dataframe,also it returns new Dataframe object after applying the function to its elements.

In [110]:
file.drop('sales_category',axis=1,inplace=True)

In [112]:
file

Item_Identifier  Item_Weight Item_Fat_Content  Item_Visibility  \
0              FDA15        9.300          Low Fat         0.016047   
1              DRC01        5.920          Regular         0.019278   
2              FDN15       17.500          Low Fat         0.016760   
3              FDX07       19.200          Regular         0.000000   
4              NCD19        8.930          Low Fat         0.000000   
...              ...          ...              ...              ...   
8518           FDF22        6.865          Low Fat         0.056783   
8519           FDS36        8.380          Regular         0.046982   
8520           NCJ29       10.600          Low Fat         0.035186   
8521           FDN46        7.210          Regular         0.145221   
8522           DRG01       14.800          Low Fat         0.044878   

                  Item_Type  Item_MRP Outlet_Identifier  \
0                     Dairy  249.8092            OUT049   
1               Soft Drinks   48.2692            OUT018   
2                      Meat  141.6180            OUT049   
3     Fruits and Vegetables  182.0950            OUT010   
4                 Household   53.8614            OUT013   
...                     ...       ...               ...   
8518            Snack Foods  214.5218            OUT013   
8519           Baking Goods  108.1570            OUT045   
8520     Health and Hygiene   85.1224            OUT035   
8521            Snack Foods  103.1332            OUT018   
8522            Soft Drinks   75.4670            OUT046   

      Outlet_Establishment_Year Outlet_Size Outlet_Location_Type  \
0                          1999      Medium               Tier 1   
1                          2009      Medium               Tier 3   
2                          1999      Medium               Tier 1   
3                          1998         NaN               Tier 3   
4                          1987        High               Tier 3   
...                         ...         ...                  ...   
8518                       1987        High               Tier 3   
8519                       2002         NaN               Tier 2   
8520                       2004       Small               Tier 2   
8521                       2009      Medium               Tier 3   
8522                       1997       Small               Tier 1   

            Outlet_Type  Item_Outlet_Sales  
0     Supermarket Type1          3735.1380  
1     Supermarket Type2           443.4228  
2     Supermarket Type1          2097.2700  
3         Grocery Store           732.3800  
4     Supermarket Type1           994.7052  
...                 ...                ...  
8518  Supermarket Type1          2778.3834  
8519  Supermarket Type1           549.2850  
8520  Supermarket Type1          1193.1136  
8521  Supermarket Type2          1845.5976  
8522  Supermarket Type1           765.6700  

[8523 rows x 12 columns]

## inplace=True permanently does the changes you made in the original Dataframe

In [115]:
file[['Item_Outlet_Sales','Item_Type','Item_Fat_Content']].groupby(['Item_Type','Item_Fat_Content']).Item_Outlet_Sales.sum()

Item_Type      Item_Fat_Content
Baking Goods   LF                   33415.8362
               Low Fat             536343.1796
               Regular             670838.1086
               low fat              10764.6544
               reg                  14163.5634
                                      ...     
Starchy Foods  LF                   22197.7720
               Low Fat             185974.5850
               Regular             126370.1716
               low fat               1618.5598
               reg                  15240.1620
Name: Item_Outlet_Sales, Length: 70, dtype: float64

In [117]:
file[['Item_MRP','Outlet_Location_Type','Outlet_Size']].groupby(['Outlet_Location_Type','Outlet_Size']).Item_MRP.mean()

Outlet_Location_Type  Outlet_Size
Tier 1                Medium         140.297699
                      Small          141.235221
Tier 2                Small          143.122481
Tier 3                High           141.425982
                      Medium         140.736687
Name: Item_MRP, dtype: float64

## groupby  function is used to split a Dataframe into groups based on some criteria and also helps to apply a function 

In [120]:
file.agg({'Item_MRP':['min','max','median'],'Item_Visibility':['mean','max']})

Item_MRP  Item_Visibility
min      31.2900              NaN
max     266.8884         0.328391
median  143.0128              NaN
mean         NaN         0.066132

## The agg() function allows user to apply a function or a list of function names  to be executed along one of the axis of the Dataframe.